File: src/main/java/com/itechart/demojavakotlin/entity/TicketEntity.java
package com.itechart.demojavakotlin.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.UUID;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "ticket")
public class TicketEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", insertable = false, updatable = false)
    private UUID id;

    @Column(name = "price", nullable = false)
    private BigDecimal price;

    @Column(name = "quantity")
    private int quantity;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private UserEntity userId;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "movie_id", nullable = false)
    private MovieEntity movieId;
}

File: kotlinpath/demojavakotlin/src/main/kotlin/by/itechart/demojavakotlin/entity/TicketEntity.kt
package by.itechart.demojavakotlin.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import java.math.BigDecimal
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "ticket")
data class TicketEntity(
        @Id
        @Column(name = "id", insertable = false, updatable = false)
        @GeneratedValue
        val id: UUID = UUID.randomUUID(),

        @Column(name = "price", nullable = false)
        val price: BigDecimal = BigDecimal(0),

        @Column(name = "quantity")
        val quantity: Int = 0,

        @ManyToOne
        @JoinColumn(name = "user_id")
        @JsonIgnore
        val userId: UserEntity? = null,

        @ManyToOne
        @JoinColumn(name = "movie_id", nullable = false)
        @JsonIgnore
        val movieId: MovieEntity
)
